### PR TITLE
DTSSTCI-75 fix code smells

### PR DESCRIPTION
**Pull request by oleon23**
### Change description ###
Fixed code smells highlighted by SonarCloud.

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSSTCI-75

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerLinkCase.java [hunk 1]
```diff
@@ -41,11 +41,15 @@ public class CaseworkerLinkCase implements CCDConfig<CaseData, State, UserRole>
     @Value("${feature.link-case.enabled}")
     private boolean linkCaseEnabled;
 
+    private final CaseLinkedNotification caseLinkedNotification;
+    
     @Autowired
-    CaseLinkedNotification caseLinkedNotification;
+    public CaseworkerLinkCase(CaseLinkedNotification caseLinkedNotification) {
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerLinkCase.java:47]: We ccan get rid of this constructor and just autowire the field

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerLinkCase.java [hunk 2]
```diff
@@ -41,11 +41,15 @@ public class CaseworkerLinkCase implements CCDConfig<CaseData, State, UserRole>
     @Value("${feature.link-case.enabled}")
     private boolean linkCaseEnabled;
 
+    private final CaseLinkedNotification caseLinkedNotification;
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerLinkCase.java:44]: Autowire this field (might need to remove the final keyword if Autowiring a final field isn't allowed)

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerMaintainLinkCase.java [hunk 3]
```diff
@@ -41,8 +41,12 @@ public class CaseworkerMaintainLinkCase implements CCDConfig<CaseData, State, Us
     @Value("${feature.link-case.enabled}")
     private boolean linkCaseEnabled;
 
+    private final CaseUnlinkedNotification caseUnlinkedNotification;
+    
     @Autowired
-    CaseUnlinkedNotification caseUnlinkedNotification;
+    public CaseworkerMaintainLinkCase(CaseUnlinkedNotification caseUnlinkedNotification) {
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerMaintainLinkCase.java:47]: Like the others I think lets remove the constructor and autowire the field

File: src/main/java/uk/gov/hmcts/sptribs/common/notification/CaseLinkedNotification.java [hunk 4]
```diff
@@ -19,11 +19,15 @@
...
 public class CaseLinkedNotification implements PartiesNotification {
 
-    @Autowired
     private NotificationServiceCIC notificationService;
 
-    @Autowired
     private NotificationHelper notificationHelper;
+    
+    @Autowired
+    public CaseLinkedNotification(NotificationServiceCIC notificationService, NotificationHelper notificationHelper) {
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/common/notification/CaseLinkedNotification.java:27]: Is sonar saying to replace autowired fields with constructors?

**Review by tomxelliott — DISMISSED** [review]

**Review by gmmagruder — DISMISSED** [review]

**Review by tomxelliott — APPROVED** [review]

**Pull request merged**